core/src/elements/kick_bomb.rs

```diff
@@ -241,13 +259,13 @@ fn update_lit_kick_bombs(
             } else if !player_standing_left && player_sprite.flip_x {
                 body.velocity.x = -kick_velocity.x;
                 body.velocity.y = kick_velocity.y;
-            } else if kick_bomb.age >= *arm_delay {
+            } else if kick_bomb.fuse_time.finished() {
                 should_explode = true;
             }
         }
 
         // If it's time to explode
-        if kick_bomb.age >= *fuse_time || should_explode {
+        if kick_bomb.arm_delay.finished() || kick_bomb.fuse_time.finished() || should_explode {
```

Comment:
This should only need to check the fuse time here:

```suggestion
        if kick_bomb.fuse_time.finished() || should_explode {
```